Framework/CurveFitting/src/Functions/ProcessBackground.cpp

```diff
@@ -415,8 +414,8 @@ void ProcessBackground::addRegion() {
   }
 
   // Check
-  for (size_t i = 1; i < vx.size(); ++i) {
-    if (vx[i] <= vx[i - 1]) {
+  for (auto it = vx.begin() + 1; it != vx.end(); ++it) {
+    if (*it <= *it - 1) {
       g_log.error()
           << "The vector X with value inserted is not ordered incrementally\n";
       throw std::runtime_error("Build new vector error!");
```

Comment:
You could use `HistogramData::isValid(...)` from  `MantidHistogramData/XValidation.h`. It works for `BinEdges`, `Points`, and `HistogramX`, but that should work here if you create `Points` already before this check, instead of below when setting the histogram.